pkg/controller/nodenetworkconfigurationpolicy/nodenetworkconfigurationpolicy_controller_test.go

```diff
@@ -0,0 +1,247 @@
+				Reconcile:    true,
+			}),
+		Entry("events with empty node selector",
+			predicateCase{
+				ObjNodeName: "node01",
+				NodeLabels: map[string]string{
+					"label1": "foo",
+					"label2": "bar",
+				},
+				NodeSelector: map[string]string{},
+				Reconcile:    true,
+			}),
+		Entry("events with matching node selector",
+			predicateCase{
+				ObjNodeName: "node01",
+				NodeLabels: map[string]string{
+					"label1": "foo",
+					"label2": "bar",
+				},
+				NodeSelector: map[string]string{
+					"label1": "foo",
+					"label2": "bar",
+				},
+				Reconcile: true,
+			}),
+		Entry("events with missing label at node",
+			predicateCase{
+				ObjNodeName: "node01",
+				NodeLabels: map[string]string{
+					"label1": "foo",
+				},
+				NodeSelector: map[string]string{
+					"label1": "foo",
+					"label2": "bar",
+				},
+				Reconcile: false,
+			}),
+		Entry("events with different label value at node",
+			predicateCase{
+				ObjNodeName: "node01",
+				NodeLabels: map[string]string{
+					"label1": "foo",
+					"label2": "bar1",
+				},
+				NodeSelector: map[string]string{
+					"label1": "foo",
+					"label2": "bar",
+				},
+				Reconcile: false,
+			}),
+		Entry("node not found",
+			predicateCase{
+				ObjNodeName:  "node02",
+				NodeLabels:   map[string]string{},
+				NodeSelector: map[string]string{},
+				Reconcile:    false,
+			}),
+	)
+})
+
+var _ = Describe("NodeNetworkConfigurationPolicy controller reconciler", func() {
+	var (
+		name              = "nodenetworkconfigurationpolicy-test"
+		numberOfNodes     = 2
+		nodeNetworkStates []nmstatev1alpha1.NodeNetworkState
+
+		cl      client.Client
+		r       *ReconcileNodeNetworkConfigurationPolicy
+		req     reconcile.Request
+		policy1 = nmstatev1alpha1.NodeNetworkConfigurationPolicy{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: name,
+			},
+		}
+		res reconcile.Result
+	)
+
+	BeforeEach(func() {
+
+		By("populate the NodeNetworkState for nodeNetworkStates")
+		nodeNetworkStates = nil
+		for n := 1; n <= numberOfNodes; n++ {
+			nodeNetworkStates = append(nodeNetworkStates, nmstatev1alpha1.NodeNetworkState{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: fmt.Sprintf("node%02d", n),
+				},
+			})
+		}
+
+		// Mock request to simulate Reconcile() being called on an event for a
+		// watched resource .
+		req = reconcile.Request{
+			NamespacedName: types.NamespacedName{
+				Name: name,
+			},
+		}
+	})
+
+	JustBeforeEach(func() {
+
+		// Register operator types with the runtime scheme.
+		By("register state and policies types")
+		s := scheme.Scheme
+		s.AddKnownTypes(nmstatev1alpha1.SchemeGroupVersion, &nmstatev1alpha1.NodeNetworkConfigurationPolicy{})
+		s.AddKnownTypes(nmstatev1alpha1.SchemeGroupVersion, &nmstatev1alpha1.NodeNetworkState{})
+
+		// Objects to track in the fake client
+		objs := []runtime.Object{&policy1}
+
+		By("create fake client")
+		cl = fake.NewFakeClient(objs...)
+
+		r = &ReconcileNodeNetworkConfigurationPolicy{client: cl, scheme: s}
+
```

Comment:
Feel free to argue, but Petr's favorite style is:

```
Foo
if err != nil {
	...
}

Bar
if err != nil {
	...
}

Foo({
	bar
	bar

	baz
})

Shrubbery
```

Keep blocks on the same level separated by empty line, keep error handling in the block where the error was returned, if starting a nested block, no need to divide it with empty line.

Reply:
Done